rocketpy/utilities.py

```diff
@@ -291,25 +290,9 @@ def _flutter_safety_factor(flight, flutter_mach):
     rocketpy.Function
         The safety factor for the fin flutter analysis.
     """
-    safety_factor = [[t, 0] for t in flutter_mach[:, 0]]
-    for i in range(len(flutter_mach)):
-        try:
-            safety_factor[i][1] = flutter_mach[i][1] / flight.mach_number[i][1]
-        except ZeroDivisionError:
-            safety_factor[i][1] = np.nan
-
-    # Function needs to remove NaN and Inf values from the source
-    safety_factor = np.array(safety_factor)
-    safety_factor = safety_factor[~np.isnan(safety_factor).any(axis=1)]
-    safety_factor = safety_factor[~np.isinf(safety_factor).any(axis=1)]
-
-    safety_factor = Function(
-        source=safety_factor,
-        inputs="Time (s)",
-        outputs="Fin Flutter Safety Factor",
-        interpolation="linear",
-    )
-
+    safety_factor = flutter_mach / flight.mach_number
+    safety_factor.set_title("Fin Flutter Safety Factor")
+    safety_factor.set_outputs("Safety Factor")
     return safety_factor
```

Comment:
I noticed that in the earlier implementation there was a check and treatment for zero division errors before making it a `Function` input. Does this more straightforward approach have the same results, because there is some treatment for `NaN` values in the `Function` division, but those vary depending on the input source type (lambda, arrays etc).

Furthermore, due to this change, I believe there should be a unit test that specifically tests these two edge cases: when there might be a `NaN` or `inf` in safety factor calculation.

Reply:
Yes the whole zerodivision thing was redundant in this code currently.
The results are simply the same.

Can't add tests as u suggested bc there's no rocket that flies on mach = 0